### PR TITLE
feat(native-filters): add sort option to select filter

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Add sort option to native select filter that defaults to true.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
The filter config modal:
image

And the filter itself:
image

### TEST PLAN
Local testing

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/ControlItems.tsx [hunk 1]
```diff
@@ -59,7 +59,10 @@ const ControlItems: FC<ControlItemsProps> = ({
           <StyledCheckboxFormItem
             key={controlItem.name}
             name={['filters', filterId, 'controlValues', controlItem.name]}
-            initialValue={filterToEdit?.controlValues?.[controlItem.name]}
+            initialValue={
+              filterToEdit?.controlValues?.[controlItem.name] ||
+              controlItem?.config?.default
```
Comment (simcha90) [inline superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/ControlItems.tsx:64]: why use here default? let's put it as default value in control panel...

File: superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx [hunk 2]
```diff
@@ -82,6 +83,7 @@ export default function PluginFilterSelect(props: PluginFilterSelectProps) {
     handleChange(currentValue ?? []);
   }, [
     JSON.stringify(currentValue),
+    sortAscending,
```
Comment (simcha90) [inline superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx:86]: why not put it in `extraFormData`? and if we not put it here so may be no need to add it this array?

Comment (villebro) [reply superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx:86]: Actually this doesn't need to be here, as this won't affect the selected value like the others in the array 👍

**Review by simcha90 — APPROVED** [review]

File: superset-frontend/src/filters/components/Select/types.ts [hunk 3]
```diff
@@ -29,10 +29,10 @@ interface PluginFilterSelectCustomizeProps {
   defaultValue?: (string | number)[] | null;
   currentValue?: (string | number)[] | null;
   enableEmptyFilter: boolean;
-  fetchPredicate?: string;
```
Comment (villebro) [inline superset-frontend/src/filters/components/Select/types.ts:32]: `fetchPredicate` is now automatically applied in backend

File: superset-frontend/src/filters/components/Select/types.ts [hunk 4]
```diff
@@ -50,7 +50,7 @@ export const DEFAULT_FORM_DATA: PluginFilterSelectCustomizeProps = {
   defaultValue: null,
   currentValue: null,
   enableEmptyFilter: false,
-  fetchPredicate: '',
   inverseSelection: false,
-  multiSelect: false,
+  multiSelect: true,
```
Comment (villebro) [inline superset-frontend/src/filters/components/Select/types.ts:54]: Enabling `multiSelect` by default to match Filter Box.

**Review by nytai — APPROVED** [review]

**Comment by junlincc** [comment]
LGTM ✅ Please merge the PR as soon as all checks pass. 🙏

**Pull request merged**